block_server.py

```diff
@@ -388,33 +387,20 @@ def _initialise_config(self, full_init=False):
         # Update Web Server text
         self.server.set_config(convert_to_json(self._active_configserver.get_config_details()))
 
-    def _stop_iocs_and_start_config_iocs(self):
-        """ Stop all IOCs and start the IOCs that are part of the configuration."""
-        # iocs_to_start, iocs_to_restart are not used at the moment, but longer term they could be used
-        # for only restarting IOCs for which the setting have changed.
-        non_conf_iocs = [x for x in get_iocs(CONTROL_SYSTEM_PREFIX) if x not in self._active_configserver.get_ioc_names()]
-        self._ioc_control.stop_iocs(non_conf_iocs)
-        self._start_config_iocs()
-
-    def _start_config_iocs(self):
+    def _start_config_iocs(self, iocs_to_start, iocs_to_restart):
         # Start the IOCs, if they are available and if they are flagged for autostart
         # Note: autostart means the IOC is started when the config is loaded,
         # restart means the IOC should automatically restart if it stops for some reason (e.g. it crashes)
-        for name, ioc in self._active_configserver.get_all_ioc_details().items():
+
+        def _should_start(ioc_name):
+            ioc = self._active_configserver.get_all_ioc_details()[ioc_name]
             if ioc.remotePvPrefix not in (None, ""):
-                print_and_log(f"IOC '{name}' is set to run remotely - not starting it.")
-                continue
+                print_and_log(f"IOC '{ioc_name}' is set to run remotely - not starting it.")
+                return False
+            return ioc.autostart
 
-            try:
-                # IOCs are restarted if and only if auto start is True. Note that auto restart instructs proc serv to
-                # restart an IOC if it terminates unexpectedly and does not apply here.
-                if ioc.autostart:
-                    if self._ioc_control.get_ioc_status(name) == "RUNNING":
-                        self._ioc_control.restart_iocs([name], reapply_auto=True)
-                    else:
-                        self.start_iocs([name])
-            except Exception as err:
-                print_and_log(f"Could not (re)start IOC {name}: {err}", "MAJOR")
+        self._ioc_control.start_iocs([ioc for ioc in iocs_to_start if _should_start(ioc)])
+        self._ioc_control.restart_iocs([ioc for ioc in iocs_to_restart if _should_start(ioc)])
 
```

Comment:
This related to comment above on changed_ioc when ioc is running. restart_iocs only gets called if ioc is supposed to autostart, so does not stop an ioc that is currently running and not set to autostart. There is a suggestion about about checking this and updating ioc_to_stop, that may be fine on its own or maybe as extra security maybe adding somethgn along the lines of 
` self._ioc_control.stop_iocs([ioc for ioc in iocs_to_restart if not _should_start(ioc)])`